pkg/tfbridge/provider.go

```diff
@@ -802,7 +804,8 @@ func (p *Provider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*pul
 		if schema != nil {
 			iv := NewInputValidator(urn, *schema)
 			typeFailures := iv.ValidateInputs(t, news)
-			if typeFailures != nil {
+			if typeFailures != nil && len(*typeFailures) > 0 {
```

Comment:
This is a code smell. It should never be the case that `typeFailures != nil && len(*typeFailures) == 0`. I would take this opportunity to change the return type of `PulumiInputValidator.ValidateInputs` to `[]TypeFailure` (from `*[]TypeFailure`).